pkg/addon/parameters_test.go

```diff
@@ -61,21 +85,39 @@ func TestGetParameter(t *testing.T) {
 					},
 				},
 			}
+			namespace := &corev1.Namespace{
+				ObjectMeta: v1.ObjectMeta{
+					Name: testOperatorNamespace,
+				},
+			}
+			subscription := &v1alpha1.Subscription{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      addonName,
+					Namespace: testOperatorNamespace,
+				},
+			}
+			if scenario.SecretFromEnvar == false {
+				namespace.Labels = labels
+				subscription.Labels = labels
+			} else {
+				os.Setenv("DEFAULT_ADDON_PARAMETERS_SECRET_NAME", "addon-managed-api-service-parameters")
```

Comment:
This worries me. In the test you are setting the a system envar. What happens when I have the operator running locally with a different envar? Also take into account I auto run the unit tests on every change I make with the code base.

Reply:
If you will be running the operator locally with a different envar it would be overridden. Override would be valid as long as you don't update [this](https://github.com/integr8ly/integreatly-operator/blob/master/config/secrets/quota-secret.yaml#L9) value. If you do update it and specify a different envar you would expect that this `os.Setenv` should be also updated :thinking: 
This test should be fine if you run units on every change as long as nothing related to the addon parameters secret gets changed. 
Would you say it might be better to check the existence of this envar before setting it?

Reply:
I am unsure to what I think the best approach is here. I think a second opinion on the matter is needed. The idea of a unit test make changes to my system does not sit well with me. When the use of envars don't seem right to me.

Reply:
Agree that setting a local env var on local system is undesirable.

> Would you say it might be better to check the existence of this envar before setting it?
yes sounds good.

Reply:
added check for existence of envar